src/components/chats/components/CreateNewChat/CreateChat.tsx

```diff
@@ -0,0 +1,38 @@
+import { FC } from 'react'
+
+import { Button } from '@chakra-ui/react'
+import { DocumentIcon } from '@heroicons/react/24/outline'
+import { useCreatePrivateChat } from '../../chatHooks'
+import { NewChatProps } from './types'
+import { useCanCreateChat } from './useCanCreateChat'
+
+export const CreateChat: FC<NewChatProps> = ({
+  postId,
+  chatId,
+  resetTextArea,
+}) => {
+  const { mutate: createChat } = useCreatePrivateChat()
+  const canCreateChat = useCanCreateChat(postId, chatId)
+
+  const handleCreateChat = () => {
+    createChat({
+      onSuccess: () => {
+        resetTextArea()
```

Comment:
This is a react antipattern. You are passing down a callback to an event. What you want is to bubble up events instead. In this case, it's just a matter of renaming `resetTextArea` to `onSuccess`. In the parent you will have `onSuccess={handleSuccess}`.

Reply:
Not sure this is an anitpattern. You mean the resetTextArea loses context with the child prop naming right?

Reply:
Here you are saying: _In order for this component to work, it needs some external code to handle the resetting of the textarea_. This is coupling, and the component cannot work independently.

If instead, you build the component saying: _Look, whenever I successfully create a chat, I will tell you in case you need to react to that event_. Then, the parent component who acts as an orchestrator reacts to the new chat being created by clearing the textarea.